lib/common/xml.c

```diff
@@ -3787,7 +3787,8 @@ dump_xml_element(xmlNode * data, int options, char **buffer, int *offset, int *m
     if (data->children) {
         xmlNode *xChild = NULL;
 
-        for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
+        //for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
+        for (xChild = data->children; xChild != NULL; xChild = xChild->next) {
```

Comment:
Yes, I deleted my previous comment once I followed the code more carefully. I commented too quickly a few times on this one :)